operator/pkg/awsprovider/instances/reconciler.go

```diff
@@ -180,24 +180,36 @@ func (c *Controller) subnetsFor(ctx context.Context, dataplane *v1alpha1.DataPla
 	if err != nil {
 		return nil, fmt.Errorf("getting subnet for %s, %w", dataplane.Spec.ClusterName, err)
 	}
-	return c.filterPrivateSubnets(ctx, subnetIDs)
+	// Returns public subnets if no private subnets are found
+	return c.filterSubnets(ctx, subnetIDs)
 }
 
-func (c *Controller) filterPrivateSubnets(ctx context.Context, ids []*string) ([]string, error) {
+func (c *Controller) filterSubnets(ctx context.Context, ids []*string) ([]string, error) {
+	if len(ids) == 0 {
+		return nil, fmt.Errorf("found zero subnets while filtering")
+	}
 	output, err := c.ec2api.DescribeSubnetsWithContext(ctx, &ec2.DescribeSubnetsInput{
 		SubnetIds: ids,
 	})
 	if err != nil {
 		return nil, fmt.Errorf("describing subnet, %w", err)
 	}
-	result := []string{}
+	public := []string{}
+	private := []string{}
 	for _, subnet := range output.Subnets {
-		if ptr.BoolValue(subnet.MapPublicIpOnLaunch) {
+		if aws.Int64Value(subnet.AvailableIpAddressCount) == 0 {
```

Comment:
Both aws/aws-sdk-go and knative/pkg have libraries for inverting and reverting pointers like this. It looks like you're swapping from relying on knative pkg to the aws sdk. But then on line 203, you're using knative/pkg again.

Reply:
Choosing to go with knative/pkg